File: java/0704_search.java
class Solution {
    public int search(int[] nums, int target) {
        int left=0,right=nums.length;
        while(left<right)
        {
            int mid = (right+left)/2;
            if(nums[mid]>=target)
                right = mid;
            else 
                left = mid+1;
        }
        return left<nums.length && nums[left]==target?left:-1;
    }
}

File: cpp/0704_search.cpp
// Binary Search：
// 初始：left=0，right=size
// 循环条件：left<right
// 更新：使用左闭右开[left,mid],(mid,right]
// 返回：如果查找得到：（1）位置一定在区间内（2）target值与该位置的值相等，反之返回-1；
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int p1=0,p2 = nums.size();
        while(p1<p2)
        {
            int mid = (p1+p2)>>1;
            if(nums[mid]>=target)
                p2 = mid;
            else
                p1 = mid+1;
        }
        return p1<nums.size() && nums[p1]==target?p1:-1;
    }
};
